src/Sulu/Component/Content/Document/Subscriber/RedirectTypeSubscriber.php

```diff
@@ -28,7 +28,7 @@ public static function getSubscribedEvents()
     {
         return [
             Events::METADATA_LOAD => 'handleMetadataLoad',
-            Events::PERSIST => 'handlePersist',
+            Events::PERSIST => ['handlePersist', 15],
```

Comment:
The redirect type must already be set on the document when the `ResourceSegmentSubscriber` is executed, because it relies on this value.